core/arch/arm/kernel/generic_boot.c

```diff
@@ -821,6 +825,30 @@ static void reset_dt_references(void)
 {
 }
 
+static void carveout_memory_range(void)
```

Comment:
Can you rename this routine to something like `discover_nsec_memory()`?

Reply:
Ok, I will change it to `discover_nsec_memory()` as your suggestion.